enzyme/Enzyme/CApi.cpp

```diff
@@ -340,6 +340,21 @@ void EnzymeRegisterAllocationHandler(char *Name, CustomShadowAlloc AHandle,
     };
 }
 
+
+/// This is the main entry point to register a custom derivative for language frontends.
+/// It should be prefered over trying to register custom-derivatives in the llvm-ir module.
+/// The main reason is that these rules can handle non-default activity cases, e.g.
+/// a function call where a pointer or a float scalar is marked as const.
+/// To get a better low-level understanding, the code in AdjointGenerator.h can be read.
+///
+/// This Function will only handle using ReverseMode AD (either split or combined).
```

Comment:
currently this only supports split mode. Well kind of. A combined mode call will call the "forward pass" generator from here and then also the "reverse pass" generator as well. This is also the case for  split mode as well. Of course with the full functionality of gradientutils/etc available, one could check if they are in combined mode by checking if gutils->mode == ReverseModeCombined.

Reply:
Can you please answer how the last three arguments should be used?